### PR TITLE
MONDRIAN-2287 - ClassCastException during native evaluation

### PR DESCRIPTION
**Pull request by YuryBY**
@lucboudreau, would you mind review/merge

### CODE REVIEW

File: src/main/mondrian/rolap/sql/CrossJoinArgFactory.java [hunk 1]
```diff
@@ -229,34 +229,38 @@ private boolean isConstrainedMeasure(Exp arg) {
...
-        Exp[] tupleArgs = getCalculatedTupleArgs(arg);
-        for (Exp tupleArg : tupleArgs) {
-            Dimension dimension = tupleArg.getType().getDimension();
-            if (!dimension.isMeasures()) {
-                List<RolapMember> members;
-                if (memberLists.containsKey(dimension)) {
-                    members = memberLists.get(dimension);
-                } else {
-                    members = new ArrayList<RolapMember>();
+        {
```
Comment (lucboudreau) [inline src/main/mondrian/rolap/sql/CrossJoinArgFactory.java]: Please revert the change in indentation.

**Comment by lucboudreau** [comment]
@mkambol Can you review please?

**Comment by mkambol** [comment]
Sure

**Comment by YuryBY** [comment]
@lucboudreau, @mkambol  - done

**Comment by mkambol** [comment]
@YuryBY  The fix looks good.  Not sure the test is adequate, though.  I broke the change by eliminating the 
  (tupleArg instanceof ResolvedFunCall) 
check, reran CrossJoinArgFactoryTest, and still saw green.

Running Will's query below fails with the expected CCE, though, and shows green with the fix.  I really like the idea of having true, mocked unit tests if you're able to do so effectively, but you should probably include minimally one test with a query and asserted results for a bug like this.

with member [Measures].[aa] as '([Measures].[Store Cost], [Gender].[M])'
  member [Measures].[bb] as '([Measures].[Store Cost], [Gender].[M].PrevMember)'
select NON EMPTY Crossjoin({[Store].[USA].[CA]}, {[Measures].[aa], [Measures].[bb]}) ON COLUMNS,
  NON EMPTY [Marital Status].[Marital Status].Members ON ROWS
from [Sales]

**Comment by lucboudreau** [comment]
@YuryBY Please see Matt's comments and add the missing tests. Thanks!

**Comment by YuryBY** [comment]
@lucboudreau, @mkambol - PR is updated
